packages/create-gator-app/templates/nextjs/delegation-starter/src/components/Steps.tsx

```diff
@@ -0,0 +1,181 @@
+"use client";
+
+import { useEffect, useState } from "react";
+import AddressCard from "@/components/AddressCard";
+import { CodeBlock } from "@/components/CodeBlock";
+import ConnectButton from "@/components/ConnectButton";
+import CreateDelegateButton from "@/components/CreateDelegateButton";
+import CreateDelegateCodeBlock from "@/components/CreateDelegateCodeBlock";
+import CreateDelegationButton from "@/components/CreateDelegationButton";
+import CreateDelegationCodeBlock from "@/components/CreateDelegationCodeBlock";
+import CreateDelegatorCodeBlock from "@/components/CreateDelegatorCodeBlock";
+import DeployDelegatorButton from "@/components/DeployDelegatorButton";
+import LearnMoreButton from "@/components/LearnMoreButton";
+import RedeemDelegationButton from "@/components/RedeemDelegationButton";
+import RedeemDelegationCodeBlock from "@/components/RedeemDelegationCodeBlock";
+import { useAccount } from "wagmi";
+import useDelegateSmartAccount from "@/hooks/useDelegateSmartAccount";
+import useDelegatorSmartAccount from "@/hooks/useDelegatorSmartAccount";
+import { useStepContext } from "@/hooks/useStepContext";
+import useStorageClient from "@/hooks/useStorageClient";
+
+export default function Steps() {
+  const { step, changeStep } = useStepContext();
+  const { address } = useAccount();
+  const { smartAccount } = useDelegatorSmartAccount();
+  const { smartAccount: delegateSmartAccount } = useDelegateSmartAccount();
+  const { getDelegation } = useStorageClient();
+  const [storedDelegation, setStoredDelegation] = useState<any>(null);
+
+  useEffect(() => {
+    if (!address) {
+      changeStep(1);
+    }
+
+    if (address && smartAccount && !delegateSmartAccount) {
+      smartAccount.isDeployed().then((isDeployed: boolean) => {
+        if (!isDeployed) {
+          changeStep(2);
+        }
+        if (isDeployed) {
+          changeStep(3);
+        }
+      });
+    }
+
+    if (address && smartAccount && delegateSmartAccount) {
+      const delegation = getDelegation(delegateSmartAccount.address);
+      if (!delegation) {
+        changeStep(4);
+      } else {
+        changeStep(5);
+      }
+    }
+  }, [address, smartAccount, delegateSmartAccount]);
+
+  // Refresh stored delegation when on step 4 or 5
+  useEffect(() => {
+    if (step === 5 && delegateSmartAccount) {
+      const delegation = getDelegation(delegateSmartAccount.address);
+      setStoredDelegation(delegation);
+    }
```

Comment:
should this be `if (step === 5 && delegateSmartAccount && !delegation)` so as to not unnecessarily re-fetch the delegation if it's already stored?

Or does this create a new delegation, so if the delegate changes, then it must recreate.

Reply:
It changes, on step 4 we create a new delegation. So on step 5, it'll always be a new delegation. We never store the delegate wallet(burner wallet) in the storage, so even if user comes back, they have to create delegate wallet, create a delegation, and then redeem it.